test/common/http/utility_test.cc

```diff
@@ -90,21 +90,22 @@ TEST(HttpUtility, createSslRedirectPath) {
   }
 }
 
-TEST(HttpUtility, parseCodecOptions) {
+TEST(HttpUtility, parseHttp2Settings) {
   {
     Json::ObjectSharedPtr json = Json::Factory::loadFromString("{}");
-    EXPECT_EQ(0UL, Utility::parseCodecOptions(*json));
+    EXPECT_EQ(0UL, Utility::parseHttp2Settings(*json).codec_options_);
   }
 
   {
     Json::ObjectSharedPtr json =
         Json::Factory::loadFromString("{\"http_codec_options\": \"no_compression\"}");
-    EXPECT_EQ(CodecOptions::NoCompression, Utility::parseCodecOptions(*json));
+    EXPECT_EQ(Http2Settings::CodecOptions::DisableDynamicHPACKTable,
+              Utility::parseHttp2Settings(*json).codec_options_);
```

Comment:
I think there's more to check here.  The other values should be set to their defaults?